src/test/kotlin/uk/gov/justice/digital/hmpps/manageusersapi/resource/UserGroupControllerIntTest.kt

```diff
@@ -169,4 +169,75 @@ class UserGroupControllerIntTest : IntegrationTestBase() {
         .expectStatus().isNoContent
     }
   }
+
+  @Nested
+  inner class AddUserGroup {
+    private val userId = UUID.fromString("7CA04ED7-8275-45B2-AFB4-4FF51432D1EC")
+    private val group = "site_1_group_2"
+
+    @Test
+    fun `access forbidden when no authority`() {
+      webTestClient.put().uri("/users/$userId/groups/$group")
+        .exchange()
+        .expectStatus().isUnauthorized
+    }
+
+    @Test
+    fun `access forbidden when no role`() {
+      webTestClient.put().uri("/users/$userId/groups/$group")
+        .headers(setAuthorisation(roles = listOf()))
+        .exchange()
+        .expectStatus().isForbidden
+    }
+
+    @Test
+    fun `access forbidden when wrong role`() {
+      webTestClient.put().uri("/users/$userId/groups/$group")
+        .headers(setAuthorisation("ITAG_USER_ADM", listOf("ROLE_AUDIT")))
+        .exchange()
+        .expectStatus().isForbidden
+    }
+
+    @Test
+    fun `adds a group to a user`() {
+      externalUsersApiMockServer.stubAddGroupToUser(userId.toString(), group)
+      webTestClient
+        .put().uri("/users/$userId/groups/$group")
+        .headers(setAuthorisation("ITAG_USER_ADM", listOf("ROLE_MAINTAIN_OAUTH_USERS")))
+        .exchange()
+        .expectStatus().isNoContent
+    }
+
+    @Test
+    fun `fail bad request`() {
+      externalUsersApiMockServer.stubAddUserGroupFail(userId.toString(), group, BAD_REQUEST)
+      webTestClient.put().uri("/users/$userId/groups/$group")
+        .headers(setAuthorisation("ITAG_USER_ADM", listOf("ROLE_MAINTAIN_OAUTH_USERS")))
+        .exchange()
+        .expectStatus().isBadRequest
+        .expectHeader().contentType(MediaType.APPLICATION_JSON)
+        .expectBody()
+        .jsonPath("$").value<Map<String, Any>> {
+          Assertions.assertThat(it["status"] as Int).isEqualTo(BAD_REQUEST.value())
```

Comment:
minor : static import for **Assertions**.assertThat

Reply:
Done